lib/confex/resolver.ex

```diff
@@ -0,0 +1,134 @@
+defmodule Confex.Resolver do
+  @moduledoc """
+  This module provides API to recursively resolve Confex configuration types in a `Map` or `Keyword` via
+  one of `Confex.Adpater`'s.
+  """
+  alias Confex.Adapter
+  alias Confex.Type
+
+  @types [:string, :integer, :float, :boolean, :atom, :module, :list]
+
+  # If you are distributing new adapter for Confex,
+  # please contribute additional value to this list.
+  @adapters Application.fetch_env!(:confex, :adapters)
```

Comment:
Module attributes are set at compilation time
So basically, confex have same problem as he plans to solve